### PR TITLE
Auth: Implement requester interface in access control module

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

Enterprise branch in progress [PR](https://github.com/grafana/grafana-enterprise/pull/5717)

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/274

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/sqlstore/permissions/dashboard.go [hunk 1]
```diff
@@ -111,14 +111,21 @@ func (f *accessControlDashboardPermissionFilter) Where() (string, []any) {
 }
 
 func (f *accessControlDashboardPermissionFilter) buildClauses() {
-	if f.user == nil || f.user.Permissions == nil || f.user.Permissions[f.user.OrgID] == nil {
+	if f.user == nil || f.user.IsNil() || f.user.GetPermissions() == nil || len(f.user.GetPermissions()) == 0 {
```
Comment (Jguer) [inline pkg/services/sqlstore/permissions/dashboard.go:114]: GetPermissions should never be nil but it doesn't hurt to check

File: pkg/services/user/identity.go [hunk 2]
```diff
@@ -46,11 +46,13 @@ func (u *SignedInUser) NameOrFallback() string {
 // TODO: There's a need to remove this struct since it creates a circular dependency
 
 // DEPRECATED: This function uses `UserDisplayDTO` model which we want to remove
+// this function uses dto to getch the gravatar url
 func (u *SignedInUser) ToUserDisplayDTO() *UserDisplayDTO {
 	return &UserDisplayDTO{
 		ID:    u.UserID,
 		Login: u.Login,
 		Name:  u.Name,
+		// AvatarURL: dtos.GetGravatarUrl(u.GetEmail()),
```
Comment (Jguer) [inline pkg/services/user/identity.go:55]: commented code doesn't match the description?

File: pkg/web/webtest/webtest.go [hunk 3]
```diff
@@ -111,9 +112,9 @@ func RequestWithWebContext(req *http.Request, c *contextmodel.ReqContext) *http.
 	return req
 }
 
-func RequestWithSignedInUser(req *http.Request, user *user.SignedInUser) *http.Request {
+func RequestWithSignedInUser(req *http.Request, usr identity.Requester) *http.Request {
```
Comment (Jguer) [inline pkg/web/webtest/webtest.go:115]: I guess you can leave this as SignedInUser if you are just going to assert it back to SignedInUser after

File: pkg/services/sqlstore/permissions/dashboard.go (hunk 1 above)
Comment (linoman) [reply pkg/services/sqlstore/permissions/dashboard.go:114]: I'm removing this in order to be consistent

File: pkg/services/user/identity.go (hunk 2 above)
Comment (linoman) [reply pkg/services/user/identity.go:55]: Added a more verbose description. Please let me know if this works for you

**Review by Jguer — APPROVED** [review]

**Pull request merged**